File: leetcode/151.reverse-words-in-a-string/cpp/reversewords_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "reversewords.cpp"

struct testCase
{
    std::string s;
    std::string want;
};

TEST_CASE("ReverseWords")
{
    std::vector<testCase> tcs{
        {"the sky is blue", "blue is sky the"},
        {"  hello world  ", "world hello"},
        {"a good   example", "example good a"},
    };

    Solution s;

    for (auto tc : tcs)
    {
        CHECK(s.reverseWords(tc.s) == tc.want);
    }
}

File: leetcode/151.reverse-words-in-a-string/ts/reversewords.test.ts
import { test, expect } from "vitest"
import reverseWords from "./reversewords.js"

interface testcase {
    s: string
    want: string
}

test("reversewords", () => {
    const tcs: testcase[] = [
        { s: "the sky is blue", want: "blue is sky the" },
        { s: "  hello world  ", want: "world hello" },
        { s: "a good   example", want: "example good a" },
    ]

    tcs.forEach(tc => expect(reverseWords(tc.s) === tc.want))
})
